zio-http/src/main/scala/zhttp/http/Request.scala

```diff
@@ -8,15 +10,16 @@ final case class Request(endpoint: Endpoint, data: Request.Data = Request.Data.e
   val route: Route          = method -> url.path
 
   def getBodyAsString: Option[String] = data.content match {
-    case HttpContent.Complete(data) => Option(data)
+      // TODO: this is strange but necessary because of how we define
+      //  `content: HttpContent[Any, Byte]` in `Request.Data`
+    case HttpContent.Complete(data) => Option(data).map(_.toString)
     case _                          => Option.empty
   }
-
 }
 
 object Request {
-  val emptyContent: HttpContent.Complete[String] = HttpContent.Complete("")
-  final case class Data(headers: List[Header], content: HttpContent[Any, String])
+  val emptyContent: HttpContent.Chunked[Any, Byte] = HttpContent.Chunked(data = ZStream.empty)
+  final case class Data(headers: List[Header], content: HttpContent[Any, Byte])
```

Comment:
I am guessing just changing the content to `HttpContent[Any, ByteBuf]` should allow us to make any kind of request. The parsing can be left to the developer or specific codecs can be provided as a part of the library.